src/crimson/osd/pg.cc

```diff
@@ -849,6 +849,45 @@ std::ostream& operator<<(std::ostream& os, const PG& pg)
   return os;
 }
 
+void PG::mutate_object(
+  ObjectContextRef& obc,
+  ceph::os::Transaction& txn,
+  osd_op_params_t& osd_op_p)
+{
+  if (obc->obs.exists) {
+    obc->obs.oi.prior_version = obc->obs.oi.version;
+    obc->obs.oi.version = osd_op_p.at_version;
```

Comment:
I don't think I understand this PR -- does this mean that we don't update version/prior_version when snap trimming?

Reply:
Um, not really, the object's version/prior_version is updated by snaptrim_event itself in `SnapTrimObjSubEvent::adjust_snaps()` and `SnapTrimObjSubEvent::update_head()`

Reply:
Tests are green. @athanatos, are you ok with merging this one?

Reply:
LGTM then